src/helm/proxy/clients/mistral_client.py

```diff
@@ -0,0 +1,131 @@
+# mypy: check_untyped_defs = False
+import requests
+from typing import Any, Dict, List
+
+from helm.common.cache import CacheConfig
+from helm.common.optional_dependencies import handle_module_not_found_error
+from helm.common.request import wrap_request_time, Request, RequestResult, Sequence, Token
+from helm.common.tokenization_request import (
+    TokenizationRequest,
+    TokenizationRequestResult,
+)
+from helm.proxy.tokenizers.tokenizer import Tokenizer
+from .client import CachingClient, truncate_sequence
+
+try:
+    from mistralai.client import MistralClient
+    from mistralai.models.chat_completion import ChatMessage, ChatCompletionResponse
+except ModuleNotFoundError as e:
+    handle_module_not_found_error(e, ["mistral"])
+
+
+class MistralAIClient(CachingClient):
+    """
+    Client for Mistral API.
+    """
+
+    # Aliases to match HELM names to Model names in Mistral API
+    _model_aliases: Dict[str, str] = {
+        "mistral-7b-v0.1": "mistral-tiny",
+    }
+
+    def __init__(self, tokenizer: Tokenizer, tokenizer_name: str, cache_config: CacheConfig, api_key: str):
+        super().__init__(cache_config=cache_config)
+        self.api_key: str = api_key
+        self.tokenizer = tokenizer
+        self.tokenizer_name = tokenizer_name
+        self._client = MistralClient(api_key=self.api_key)
+
+    def _send_request(self, model_name: str, raw_request: Dict[str, Any]) -> Dict[str, Any]:
+        messages = [ChatMessage(role="user", content=raw_request["prompt"])]
+
+        chat_response: ChatCompletionResponse = self._client.chat(
+            model=model_name,
+            messages=messages,
+            temperature=raw_request["temperature"],
+            max_tokens=raw_request["max_tokens"],
+            top_p=raw_request["top_p"],
+            random_seed=raw_request["random_seed"],
+            safe_prompt=False,  # Disable safe_prompt
+        )
+
+        result = {
+            "choices": [
+                {
+                    "text": choice.message.content
+                    if choice.message.content and choice.message.role == "assistant"
+                    else "",
+                }
+                for choice in chat_response.choices
+            ],
+        }
+
+        return result
+
+    def make_request(self, request: Request) -> RequestResult:
+        """Make a request"""
+        raw_request = {
+            "prompt": request.prompt,
+            "max_tokens": request.max_tokens,
+            "temperature": request.temperature,
+            "top_p": request.top_p,
+            "random_seed": request.random,
```

Comment:
We discussed offline briefly about the interaction between `random_seed` and `num_completions`... let's deal with it in a separate pull request, since I'd rather to get the Mistral client in ASAP.

Reply:
I can create a PR on that yes